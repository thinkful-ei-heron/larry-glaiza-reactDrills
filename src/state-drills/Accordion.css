ul {
    background-color: lightgray;
    width: 50%;
    margin-left:auto;
    margin-right: auto;
    list-style:none;
    padding: 2px;
  }
  
  button {
    padding: 5px;
    margin: 3px;
    font-size: 1rem;
  }

  li {
    text-align: center;
    color: #fff;

  }
  
  button:hover {
    background-color: green;
  }
  
  p{
    width: 90%;
    border: 1px solid black;
    margin-left:auto;
    margin-right:auto;
  }