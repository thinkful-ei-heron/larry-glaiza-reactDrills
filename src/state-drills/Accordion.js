import React, { Component } from 'react';
import './Accordion.css';

export class Accordion extends Component {
    static defaulProps = {
        sections: []
    };

    state = {
        lastClicked: null,
    }

    handleClick = (index) => {
        this.setState( {
            lastClicked: index
        })
    }

    displayList = (item, index) => {
        return (
            <li key={index} >
                <button onClick={() => this.handleClick(index)}> {item.title}</button>
                {(this.state.lastClicked === index) && <p>{item.content}</p>}
            </li>
        )
    }
        
    render() {
        return (
            <ul>
                {this.props.sections.map((item, index) => 
                    this.displayList(item, index)
                )}
            </ul>
            )
          }
}
        

export default Accordion;
