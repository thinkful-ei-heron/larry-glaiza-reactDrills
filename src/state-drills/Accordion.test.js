import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Accordion from './Accordion';

const  sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
]

  describe('<ul>', () => {
    it('renders empty list', () => {
        const result = renderer.create(<Accordion />).toJSON();
        expect(result).toMatchSnapshot(); 
    });

  });