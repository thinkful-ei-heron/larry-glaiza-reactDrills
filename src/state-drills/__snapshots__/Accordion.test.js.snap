// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ul> Only one content shown 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
       
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
       
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
       
      Section 3
    </button>
  </li>
</ul>
`;

exports[`<ul> Opens contents when clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
       
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
       
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
       
      Section 3
    </button>
  </li>
</ul>
`;

exports[`<ul> renders no sections as active by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
       
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
       
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
       
      Section 3
    </button>
  </li>
</ul>
`;
